runners/direct-java/src/test/java/org/apache/beam/runners/direct/portable/ReferenceRunnerTest.java

```diff
@@ -101,4 +104,29 @@ public void process(@Element KV<String, Iterable<Integer>> e,
             PipelineOptionsTranslation.toProto(PipelineOptionsFactory.create()));
     runner.execute();
   }
+
+  @Test
+  public void testGBK() throws Exception {
+    Pipeline p = Pipeline.create();
+
+    PAssert.that(
+            p.apply(Create.of(KV.of(42, 0), KV.of(42, 1), KV.of(42, 2)))
+                // Will create one bundle for each value, since direct runner uses 1 bundle per key
+                .apply(Reshuffle.viaRandomKey())
+                // Multiple bundles will emit values onto the same key 42.
+                // They must be processed sequentially rather than in parallel, since
+                // the trigger firing code expects to receive values sequentially for a key.
+                .apply(GroupByKey.create())
+                .apply(Values.create())
+                .apply(Flatten.iterables()))
```

Comment:
If the grouping didn't happen correctly we would still produce the same result and pass the test. We should drop Flatten.iterables, better to validate the output was the an iterable containing [0, 1, 2].

Reply:
Done.